### PR TITLE
Feature/implement mixins

### PR DESCRIPTION
**Pull request by willread**
## Description

Implement the new shared mixins from `rise-common-component`.

## Motivation and Context

Work related to [this Trello card](https://trello.com/c/GiARDdSl).

- Remove references to `image-format-invalid` and `image-rls-error` from docs, as these are now handled by `ValidFilesMixin` and `WatchFilesMixin` respectively
- Update `rise-common-component` to latest version
- Implement `WatchFilesMixin` and deprecate unneeded code
- Implement `ValidFilesMixin` and deprecate unneeded code
- Rename `_isValidFiles` to `_isValidFilesString` to prevent conflict with `ValidFilesMixin` and clarify functionality
- Update all unit tests as needed (some tests which are duplicative of tests in `rise-common-component` are removed)

## How Has This Been Tested?

Unit tests have been updated and and run against the updated component.

Additionally, the new `rise-image` component code has been deployed to [GCS here](https://widgets.risevision.com/staging/components/rise-image/2019.09.03.08.37/rise-image.js) and a test version of the `example-pud-image` template has been pushed to `stable` which points to that code.

An example presentation using `example-pud-image` has been created [here](https://apps.risevision.com/templates/edit/eda5aeb5-e52b-48e4-8afb-f6175c033b6a/?cid=7fa5ee92-7deb-450b-a8d5-e5ed648c575f) and added to [this schedule](https://apps.risevision.com/schedules/details/74b306df-7e23-485a-8d4d-5b14f818ae71?cid=7fa5ee92-7deb-450b-a8d5-e5ed648c575f).

Several different players were added to this schedule and I tested that things behaved as expected when adding or removing images, and that PUD was triggered as expected and the schedule moved through several different presentations.

Additionally, an updated demo template which points to this code was deployed [here](https://widgets.risevision.com/staging/pages/2019.09.03.08.39/src/rise-image.html) and added to the schedule to ensure that branding logos and other features demonstrated in the demo page were functioning correctly.

## Release Plan:
- As the Submitter, upon requesting review of this pull request, I confirm that the [Release Checklist](https://help.risevision.com/hc/en-us/articles/360031119991) has been completed. 
- As the Reviewer, upon approving the changes in this PR, I confirm I have reviewed and I agree that the [Release Checklist](https://help.risevision.com/hc/en-us/articles/360031119991) has been completed

#### Release Checklist Items Skipped?

No release checklist items were skipped.

### CODE REVIEW

**Comment by willread** [comment]
@santiagonoguez @stulees If possible I'd like both of you to look at this as it's a lot of changes and I haven't really worked on `rise-image` before so I want to make sure everything is solid.

I'm also a little concerned that we'll now need to maintain both components if we find any issues related to any of this code during our last round of `rise-video` QA, so I thought I'd suggest maybe blocking this PR until we've had some more time to test `rise-video`, but let me know what you guys think!

File: README.md [hunk 1]
```diff
@@ -93,9 +93,7 @@ The component may log the following:
 - **_image-reset_** ( info ): The component observed changes to either _files_ or _duration_ attributes and performs a complete reset to use latest values.
 - **_image-svg-usage_** ( info ): Provides an SVG file _blob size_ and _data url length_ info for investigative purposes.
 - **_image-load-fail_** ( error ): When attempting to render an available image, the image load failed.
-- **_image-format-invalid_** ( error ): A GCS path was set that targets a file with an invalid image file format. Valid image file formats are: jpg, jpeg, png, bmp, svg, gif, and webp.
 - **_image-svg-fail_** ( error ): When component is targeting an SVG file, the component converts the local file URL to a data url to support running on Electron Player. This error event indicates the attempt to get data url or render the SVG file failed.
-- **_image-rls-error_** ( error ): An error is received from Rise Local Storage for a file
```
Comment (santiagonoguez) [inline README.md]: @willread I think we still need to provide links to the events that the mixins can log ( maybe references to the other readme file ).

Comment (willread) [reply README.md]: @santiagonoguez Good idea, I'll add links to the mixins each component uses.

**Comment by willread** [comment]
> @willread one first comment, but let me give it a more in depth review.
> 
> I'm wondering if this, as it's a not trivial refactoring to a component that now has been widely deployed, would benefit from being deployed using and approach similar to what @stulees did for this latest rise-data-financial changes ( updated as higher major version, release for a single template, monitor, and if ok go back to previous version and deploy widely ). Thoughts ?

@santiagonoguez I think that's a good idea, provided there's a component with low usage we could use.

**Comment by stulees** [comment]
@willread @santiagonoguez I would also add that once you've deployed, validation should be on your own displays. Whatever user displays that are running the template should be left to do their scheduled reboot (commonly around 2am), so validation should also happen the next day for those displays (logs, display screenshot).

**Comment by willread** [comment]
@stulees I'm leaning towards waiting to merge this until after your Issue #17 fix  so it's not blocked by this. I can leave the test template I set up for this running on my displays in the meantime and keep an eye out for any errors / issues. What do you think?

**Comment by willread** [comment]
@santiagonoguez The `fall-is-here` template looks to be used on only 5 displays, so I think that's a good choice for testing. Does that make sense to you? I'll verify the usage again before I push the changes to the template.

**Comment by stulees** [comment]
@willread Yes makes sense, issue #17 is going to be worked on very shortly. Cheers

**Pull request merged**